### PR TITLE
pull request: new welcome page

### LINKED ISSUE #1: Deployment Environment
#### As a developer I want a continuous deployment environment setup for my new Rails application.

Acceptance criteria:
- [x] [GitHub](https://github.com/AppFolioOnboarding) is used for source
  control.
- [x] The project's version of both Ruby and Rails are used both locally and on
  Heroku.
- [x] The landing page, `/`, provides some content.
- [x] Every available controller action has corresponding tests.
- [x] [CircleCI](https://circleci.com/) is used for Continuous Integration.
- [x] Heroku is used to deploy the application.
- [x] Pull requests are automatically deployed as review applications.
- [x] The staging application automatically deploys whenever the master branch
  is updated.
- [x] The staging application automatically deploys only after all tests pass
  on CircleCI.
- [x] The staging application deployment can be promoted to the production
  application deployment.
- [x] [Pronto](https://github.com/mmozuras/pronto) is used to help with code
  style consistency.

Discussion Topics:
- [ ] Where can we discover how long it takes to return an HTTP response?
- [x] When should we commit our code?

References:
* https://rvm.io/rvm/install#try-out-your-new-rvm-installation
* http://guides.rubyonrails.org/getting_started.html
* http://guides.rubyonrails.org/testing.html#functional-tests-for-your-controllers
* https://devcenter.heroku.com/articles/getting-started-with-rails4
* https://devcenter.heroku.com/articles/pipelines
* https://github.com/bbatsov/ruby-style-guide

### PR DESCRIPTION
**Pull request by mqchau**
Closes #1

### CODE REVIEW

**Comment by bboe** [comment]
Is this ready for review? If so can you add the "waiting on reviewer" label?

Also now that the issue has an associated pull request, can you please change issue #1's label to "pull request".

**Comment by bboe** [comment]
Awesome. Can you also remove the "in progress" label on the issue? Once it has the "pull request" label the status of the issue defers to the pull request.

File: app/assets/javascripts/welcome.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+// Place all the behaviors and hooks related to the matching controller here.
```
Comment (bboe) [inline app/assets/javascripts/welcome.js]: Is this file necessary?

File: app/assets/stylesheets/welcome.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Welcome controller here.
```
Comment (bboe) [inline app/assets/stylesheets/welcome.scss]: Is this file necessary?

File: app/helpers/welcome_helper.rb [hunk 3]
```diff
@@ -0,0 +1,2 @@
+module WelcomeHelper
```
Comment (bboe) [inline app/helpers/welcome_helper.rb]: Is this file necessary?

File: config/database.yml [hunk 4]
```diff
@@ -22,4 +22,5 @@ test:
 
 production:
   <<: *default
+  # WHY ARE WE USING SQLITE3 FOR PRODUCTION?
```
Comment (bboe) [inline config/database.yml]: Does this comment add much?

To answer your question sqlite is not being used in production. Heroku replaces this file with on that works for it which ends up using postgres.

File: config/routes.rb [hunk 5]
```diff
@@ -1,3 +1,5 @@
 Rails.application.routes.draw do
+  root 'welcome#index'
+  get 'welcome/index'
```
Comment (bboe) [inline config/routes.rb]: Is this route necessary?

File: test/controllers/welcome_controller_test.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+require 'test_helper'
+
+class WelcomeControllerTest < ActionDispatch::IntegrationTest
+  test 'should get index' do
+    get welcome_index_url
+    assert_response :success
+    assert_match 'Welcome', @response.body
```
Comment (bboe) [inline test/controllers/welcome_controller_test.rb]: Can you do this with `assert_select`? We expect the body to be HTML so we can be a little more precise. `assert_match` would be good if you expect an unstructured text output.

**Comment by bboe** [comment]
Great start to this issue.

Can you please address the comments I have made?

File: config/routes.rb (hunk 5 above)
Comment (mqchau) [inline config/routes.rb]: 1st route is necessary to display the page in the beginning
2nd route is necessary to run welcome_controller_test to define variable welcome_index_url

**Comment by mqchau** [comment]
I fixed what you recommended, Bryce.

File: config/routes.rb (hunk 5 above)
Comment (bboe) [inline config/routes.rb]: Can you use `root_url` in the welcome controller test?

**Comment by bboe** [comment]
Almost perfect. Can you squash this into three commits as there are three separate atomic commits? (1) Feature addition, (2) rubocop fix, (3) seed fix.

**Comment by mqchau** [comment]
After I squashed the commits into 3 commits, I can't push to the same branch, right? What should I do after I squash?

**Comment by mqchau** [comment]
Bryce, do you or I merge this pull request? I think it's ready.

File: config/routes.rb (hunk 5 above)
Comment (bboe) [inline config/routes.rb]: Did you see the above comment?

File: test/controllers/welcome_controller_test.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
+require 'test_helper'
+
+class WelcomeControllerTest < ActionDispatch::IntegrationTest
+  test 'should get index' do
+    get welcome_index_url
+    # check response code 200
```
Comment (bboe) [inline test/controllers/welcome_controller_test.rb]: Do these comments add any value?

File: test/controllers/welcome_controller_test.rb [hunk 8]
```diff
@@ -0,0 +1,13 @@
...
+    assert_response :success
+    # check first heading of the page has Welcome string in it
+    assert_select 'body h1' do |h1|
```
Comment (bboe) [inline test/controllers/welcome_controller_test.rb]: There is a way to use `assert_select` to match the text. Can you get that to work?

Also, do we care if the h1 is really under a body element? In HTML5 `<body>` tags aren't actually required.

**Comment by bboe** [comment]
@mqchau commits look good. I had a few more comments which it would be great if you took a look at. Feel free to ask me directly if you'd like some clarification.

**Comment by mqchau** [comment]
Fixed as you said

**Comment by bboe** [comment]
Great job!
